### PR TITLE
Add existing global attribute layout

### LINKED ISSUE #36661: Variable product: Add existing attribute
Using the old experience.
It'd be great to have a more focused empty state when there is at least one attribute available.

Design: `6cBYbZmiet1l7gzc28ce8k-fi-5097%3A253868&t=jQo137F0guQ0Rqw0-4`.

When the user selects an attribute, they land in the edit form (`6cBYbZmiet1l7gzc28ce8k-fi-5096%3A253867&t=jQo137F0guQ0Rqw0-4`).

image

From issue [#35115](https://github.com/woocommerce/woocommerce/issues/35115). More context [here](https://github.com/woocommerce/woocommerce/issues/35115#issuecomment-1335707886).

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

This PR adds a new layout to add variations for global attributes. It also allows the creation of new attributes.

When there is at least one global attribute available and no local one.

Screen Capture on 2023-02-24 at 13-50-46

Design: 6cBYbZmiet1l7gzc28ce8k-fi-5097%3A253868&t=jQo137F0guQ0Rqw0-4.

When the user selects an attribute, they land in the edit form (6cBYbZmiet1l7gzc28ce8k-fi-5096%3A253867&t=jQo137F0guQ0Rqw0-4).

Closes #36661.

### How to test the changes in this Pull Request:

1. In a brand new site. Checkout this branch
2. Go to `Products > Add New > Product data > Attributes`. You should see this form:

3. Go to `Product > Attributes` and create an attribute.
4. Go back to `Products > Add New > Product data > Attributes`.
5. The following form will be visible:

Screenshot 2023-02-24 at 13 22 37

6. Press `Create new attribute` or press the selector and choose the attribute you just created.
7. After that the edit form should be visible:

Screenshot 2023-02-24 at 13 25 14

A follow-up PR will tackle loading the variations under `Value(s)`.

8. Remove the attribute you just added by pressing `Remove`.
9. The form we saw in step 5 will be visible again. The save button won't be visible.
10. Create a new attribute with variations and save it.

### Other information:

-   [x] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [x] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?
-   [x] Have you included testing instructions?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-attributes.php [hunk 1]
```diff
@@ -1,61 +1,89 @@
...
 }
+
+global $wc_product_attributes;
+// Array of defined attribute taxonomies.
+$attribute_taxonomies = wc_get_attribute_taxonomies();
+// Product attributes - taxonomies and custom, ordered, with visibility and variation attributes set.
+$product_attributes              = $product_object->get_attributes( 'edit' );
+$has_local_attributes            = empty( $attribute_taxonomies );
+$has_global_attributes           = empty( $product_attributes );
+$is_add_glboal_attribute_visible = ! $has_local_attributes && $has_global_attributes;
```
Comment (louwie17) [inline plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-attributes.php]: Spelling mistake: `glboal` -> `global`

**Review by louwie17 — CHANGES_REQUESTED** [review]
Nice work @octaedro, this was testing quite well and changes look good.
I did notice two very small things, a spelling mistake ( in code comment ) and the `Expand / close` buttons show up for me on the bottom of the `Add descriptive pieces of information...` screen (see screenshot )
I think we did want to hide this as well, maybe just hide the whole toolbar as we are already hiding the `Save Attributes` button.
Screenshot 2023-02-27 at 4 10 48 PM

File: plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-attributes.php (hunk 1 above)
Comment (octaedro) [reply plugins/woocommerce/includes/admin/meta-boxes/views/html-product-data-attributes.php]: Nice catch!

**Comment by octaedro** [comment]
Thank you @louwie17 for your review.
I just addressed the changes that you suggested. Could you give this PR another look?

**Review by louwie17 — CHANGES_REQUESTED** [review]
Thanks for updating the changes, I did notice one more bug which may of been caused by one of my prior suggestions (I am not sure).
When I don't have any global attributes and remove the auto added one the attributes section is fully blank, not allowing me to add any new attributes anymore.
See video:

https://user-images.githubusercontent.com/2240960/221814475-70a57be0-b05e-4418-9e0b-7f299e7418a3.mp4

**Comment by octaedro** [comment]
> When I don't have any global attributes and remove the auto added one the attributes section is fully blank, not allowing me to add any new attributes anymore.

Nice catch @louwie17! I fixed it in the commit [552e120](https://github.com/woocommerce/woocommerce/pull/36944/commits/552e120e53b6a4e554b4160014742a125f239031)

**Review by louwie17 — CHANGES_REQUESTED** [review]
Hey Fernando, sorry, but I found some more edge cases :| as I was playing around with this.
But you did fix the last issue, thanks!
I know there is a lot of funny JQuery logic going on in here, so rather tough to find all the use cases in one go.

Edge case 1:
- Add a global attribute and add some terms and click **Save attributes**
- Remove the just added global attributes
- Add the same global attribute back in and add terms
- Click **Save attributes** and notice how it shows two rows of the global attribute
Screenshot 2023-03-01 at 9 41 27 AM
- Click **Remove** on one and notice the placeholder is back but it shows the other global attribute at the bottom.
Screenshot 2023-03-01 at 9 42 16 AM


Edge case 2:
- Add a global attribute through the new dropdown
- Don't add any terms and click **Save attributes**
- Only the **Save attributes** button shows
Expected: show either the toolbar again at the top, or the placeholder screen with the dropdown.
Screenshot 2023-03-01 at 9 37 10 AM

**Comment by octaedro** [comment]
Thank you @louwie17 for reporting these bugs.
I added [a small fix](https://github.com/woocommerce/woocommerce/pull/36944/commits/7d4a250a817cb74f59c0bdcb3423b66517bd306f) for [this screen](https://user-images.githubusercontent.com/2240960/222088340-d0617287-a434-4da0-9785-1c3477ab4c14.png). We should add a fix [like this](https://github.com/woocommerce/woocommerce/blob/3e2443713ca9debc158999bad3c23c965fe3b713/plugins/woocommerce/client/legacy/js/admin/meta-boxes-product-variation.js#L62-L64) to solve the 2nd case. (More context for that behavior [here](https://github.com/woocommerce/woocommerce/pull/36957#issuecomment-1450001696)).
Since the errors you mentioned are older than this PR, maybe we can merge this and create a couple of issues to tackle the bugs you mentioned in a couple of follow-up PRs.

**Review by louwie17 — APPROVED** [review]
Thanks for fixing that second scenario, and you are right that the first scenario is already present in `trunk`.
@octaedro do you know if there is an issue already created for that? So we can fix that.

For the rest this works as expected, thanks for sticking with the updates 🎉

**Pull request merged**